gen/gen.go

```diff
@@ -0,0 +1,475 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"log"
+	"os"
+	"path/filepath"
+	"reflect"
+	"regexp"
+	"sort"
+	"strings"
+	"text/template"
+
+	"github.com/go-viper/mapstructure/v2"
+	mondoov1 "go.mondoo.com/mondoo-go"
+)
+
+func main() {
+	// Generate resources for Mondoo integrations only
+	err := generateIntegrationResources()
+	if err != nil {
+		log.Fatalln(err)
+	}
+}
+
+type IntegrationResource struct {
+	ResourceClassName     string
+	TerraformResourceName string
+	Fields                map[string]Field
+}
+
+func NewField(base Field, raw any) Field {
+	base.RawStruct = raw
+	return base
+}
+
+type Field struct {
+	RawStruct           any
+	GoType              string
+	MondooType          string
+	TerraformType       string
+	TerraformSchemaType string
+	TerraformSubType    string
+	HclType             string
+	GoFmtVerb           string
+}
+
+func (f Field) GoTestValue(name string, testcase int) string {
+	switch f.MondooType {
+	case BooleanField.MondooType:
+		if testcase%2 == 0 {
+			return "true"
+		}
+		return "false"
+	case StringField.MondooType:
+		return fmt.Sprintf("\"%s_%d\"", name, testcase)
+	case StringPtrField.MondooType:
+		return fmt.Sprintf("\"%s_%d\"", name, testcase)
+	case ArrayStringPtrField.MondooType:
+		return fmt.Sprintf("[]string{\"%s_%d\"}", name, testcase)
+	}
+	return "\"unimplemented: check gen/gen.go\""
+}
+
+func (f Field) ConfigurationOption(name string) string {
+	switch f.MondooType {
+	case BooleanField.MondooType:
+		return fmt.Sprintf("mondoov1.Boolean(m.%s.ValueBool())", name)
+	case StringField.MondooType:
+		return fmt.Sprintf("mondoov1.String(m.%s.ValueString())", name)
+	case StringPtrField.MondooType:
+		return fmt.Sprintf("mondoov1.NewStringPtr(mondoov1.String(m.%s.ValueString()))", name)
+	case ArrayStringPtrField.MondooType:
+		return fmt.Sprintf("ToPtr(ConvertSliceStrings(m.%s))", name)
+	}
+	return "\"unimplemented: check gen/gen.go\""
+}
+
+func (f Field) ImportConvertion(resourceClassName, fieldName string) string {
+	attr := fmt.Sprintf("integration.ConfigurationOptions.%sConfigurationOptions.%s", resourceClassName, fieldName)
+	switch f.MondooType {
+	case BooleanField.MondooType:
+		return fmt.Sprintf("types.BoolValue(%s)", attr)
+	case StringField.MondooType:
+		return fmt.Sprintf("types.StringValue(%s)", attr)
+	case StringPtrField.MondooType:
+		return fmt.Sprintf("types.StringPointerValue(%s)", attr)
+	case ArrayStringPtrField.MondooType:
+		return fmt.Sprintf("ConvertListValue(%s)", attr)
+	}
+	return "\"unimplemented: check gen/gen.go\""
+}
+
+func (f Field) AttributeOptionalOrRequired(name string) string {
+	rawField, ok := findField(f.RawStruct, name)
+	if !ok {
+		panic("field in struct not found")
+	}
+
+	tag := parseTFGenTag(rawField.Tag.Get("tfgen"))
+	if tag.Required {
+		return "Required: true"
+	}
+	return "Optional: true"
+}
+
+type tfgenTag struct {
+	Required    bool
+	Description string
+}
+
+func parseTFGenTag(tag string) tfgenTag {
+	t := tfgenTag{}
```

Comment:
it might be better to use the functions provided in the `reflect` package
https://pkg.go.dev/reflect#StructTag.Get

Reply:
I am doing that already, this function is not getting the tag from the field, this function is parsing the tag that could contain multiple values.